File: _examples/nextjs-postgres/scripts/restart-and-wait.mjs
import { execFileSync } from 'node:child_process';
import { setTimeout } from 'node:timers/promises';

const service = process.argv[2];
const url = process.env.APP_URL;
if (!service || !url) throw new Error('Set APP_URL and pass the service name');

async function health() {
  const response = await fetch(`${url}/health`, { signal: AbortSignal.timeout(5000), cache: 'no-store' });
  if (!response.ok) throw new Error(`Health returned ${response.status}`);
  const body = await response.json();
  if (body.ready !== true || typeof body.instanceId !== 'string' || !body.instanceId) {
    throw new Error('Health must include ready and instanceId');
  }
  return body.instanceId;
}

const before = await health();
execFileSync('lizard', ['restart', '--service', service, '--json'], { stdio: 'inherit', timeout: 30000 });
const deadline = Date.now() + 120000;
let candidate;
let stable = 0;
while (Date.now() < deadline) {
  try {
    const current = await health();
    stable = current !== before ? (current === candidate ? stable + 1 : 1) : 0;
    candidate = current;
    if (stable >= 3) {
      console.log(`Replacement process ready: ${before} -> ${current}`);
      process.exit(0);
    }
  } catch { stable = 0; }
  await setTimeout(2000);
}
throw new Error('No stable replacement process within 120 seconds; inspect lizard events and logs');

File: _examples/langgraph-postgres/restart_and_wait.py
import json
import os
import subprocess
import sys
import time
from urllib.error import URLError
from urllib.request import Request, urlopen

if len(sys.argv) != 2 or not os.environ.get("APP_URL"):
    raise SystemExit("Set APP_URL and pass the service name")


def health():
    request = Request(os.environ["APP_URL"] + "/health", headers={"Cache-Control": "no-cache"})
    with urlopen(request, timeout=5) as response:
        body = json.load(response)
    if body.get("ready") is not True or not isinstance(body.get("instanceId"), str) or not body["instanceId"]:
        raise ValueError("Health must include ready and instanceId")
    return body["instanceId"]


before = health()
subprocess.run(["lizard", "restart", "--service", sys.argv[1], "--json"], check=True, timeout=30)
deadline = time.monotonic() + 120
candidate, stable = None, 0
while time.monotonic() < deadline:
    try:
        current = health()
        stable = (stable + 1 if current == candidate else 1) if current != before else 0
        candidate = current
        if stable >= 3:
            print(f"Replacement process ready: {before} -> {current}")
            break
    except (URLError, TimeoutError, ValueError, OSError):
        stable = 0
    time.sleep(2)
else:
    raise SystemExit("No stable replacement process within 120 seconds; inspect lizard events and logs")
